app/uk/gov/hmrc/mtdsatestsupportapi/models/domain/VendorClientId.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mtdsatestsupportapi.models.domain
+
+case class VendorClientId(id: String) {
+  require(VendorClientId.isValid(id), s"$id is not a valid id.")
+}
+
+object VendorClientId {
+  def isValid(id: String): Boolean = id != null // TODO:  add further additional validation we chose to do (e.g. regex matching)
```

Comment:
Checking if something equals `null` [feels wrong](https://stackoverflow.com/questions/23992051/best-practice-for-null-checking-in-scala) (although I realise we're currently doing this in the nino model). 

Are we sure that we need to do any validation for `vendorId` anyway? The client_Id is provided by the platform - so if the vendor supplies an invalid id he would be told about it before the request ever reaches us.

Reply:
I just put then null check there as a placeholder, but yes may not need the validation.  I will remove it